etcd/v3/kv_etcd.go

```diff
@@ -803,43 +807,63 @@ func (et *etcdKV) watchStart(
 	if waitIndex != 0 {
 		opts = append(opts, e.WithRev(int64(waitIndex+1)))
 	}
+	session, err := concurrency.NewSession(et.kvClient, concurrency.WithTTL(defaultSessionTimeout))
```

Comment:
Probably the client is fixed in v3.2.0 
I am following up with etcd here - https://github.com/coreos/etcd/issues/7941

And if that works then we might not need this change at all.